### ISSUE TITLE
Trigger ALL sticky events on registerSticky()

### ISSUE THREAD

**Issue by ghost** [greenrobot/EventBus#90]
It would be great if registerSticky() method trigger ALL sticky events current class is subscribed to. My case: activity (sender), and fragment (built into layout) inside it (receiver). Activity sends bunch of events from onCreate, but, of course, they should be posted as sticky as there still no live fragment.
When fragment is fully created it is registered to the EventBus, but now I have to check if there are sticky events for ALL event types this fragment is subscribed to. Very ugly.
It would be much better if registerSticky() trigger delivery of all sticky events sent to current class.

**Comment by nunocm** [comment]
I agree, It would be useful.

**Comment by qianzui** [comment]
I agree too! If  only the last sticky event delivered to the subscriber, it means sometimes maybe miss some useful event!

**Comment by C0nstan** [comment]
+1

**Comment by miroslavign** [comment]
Meaning, keeping a queue of sticky event per event class. 
And then, we would need, methods for 
1) clearing this queue (per event class)
2) get all events
3) get last event

Does anyone knows of a fork based on version 3 that implemented this ?

**Comment by miroslavign** [comment]
Or maybe, just keep a counter of how many sticky events were fired, until 
a subscriber subscribed to this event class instance. 
So, saving only the last one, BUT knowing there were more, a user can decide
to take some extra steps in, say, refreshing a view.

**Comment by greenrobot-team** [comment]
Sounds to me like you need proper persistence (a queue), like a log file or database. Or as said above, create an in-memory queue that you access if required. -ut

**Comment by miroslavign** [comment]
Yes, i started a few commits on my fork. After i create something that works, I will post it here, or create a pull request. So If it is in compliance with your code, other may benefit from it.
